From ad71faa6da0916f1520918b180d88fbea3816c75 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-ad71faa6da-20210727T000657Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sat, 24 Jul 2021 18:30:26 +0200
Subject: [PATCH] http.c: use error_errno(), not error() after fopen() failure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Improve the error reporting added in e929cd20bb3 (http.c: new
functions for the http API, 2009-06-06) to emit strerror(), as fopen()
is a failing system call we'll have a meaningful errno to report.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 http.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/http.c b/http.c
index 8119247149..f7940f1b5e 100644
--- a/http.c
+++ b/http.c
@@ -2341,8 +2341,8 @@ struct http_pack_request *new_direct_http_pack_request(
 	strbuf_addf(&preq->tmpfile, "%s.temp", sha1_pack_name(packed_git_hash));
 	preq->packfile = fopen(preq->tmpfile.buf, "a");
 	if (!preq->packfile) {
-		error("Unable to open local file %s for pack",
-		      preq->tmpfile.buf);
+		error_errno("Unable to open local file %s for pack",
+			    preq->tmpfile.buf);
 		goto abort;
 	}
 
-- 
2.32.0.988.g1a6a4b2c5f

